src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -707,10 +708,54 @@ public boolean addRecipe(Recipe recipe)
 	}
 
 	@Override
-	public List<Recipe> getRecipesFor(ItemStack result)
+	public List<Recipe> getRecipesFor(@NotNull ItemStack item)
 	{
 		assertMainThread();
-		return recipes.stream().filter(recipe -> recipe.getResult().equals(result)).collect(Collectors.toList());
+
+		return recipes.stream().filter(recipe -> {
+			ItemStack result = recipe.getResult();
+			// Amount is explicitly ignored here
+			return result.getType() == item.getType() && result.getItemMeta().equals(item.getItemMeta());
+		}).collect(Collectors.toList());
```

Comment:
I know this PR is already closed but I just wanted to recommend that you replace these lines with something like this.
I haven't tested it, but I think you should use ItemStack#isSimilar instead of doing your own comparison
```
return recipes.stream()
	.map(Recipe::getResult)
	.filter(item::isSimilar)
	.collect(Collectors.toList());
```

Reply:
Oh right, I forgot that .isSimilar() ignores the amount too

Reply:
This would return a `List<ItemStack` though instead of a `List<Recipe>`.
Perhaps
```java
return recipes.stream()
    .filter(item -> item.isSimilar(item.getResult()))
    .collect(Collectors.toList());
```

Reply:
```
return recipes.stream()
    .filter(recipe -> item.isSimilar(recipe.getResult()))
    .collect(Collectors.toList());
```

Reply:
This would of course result in a NPE if item is null, is this inline with Spigot behaviour too?

Reply:
Oh nevermind, it is marked as NonNull anyway, looks good!